web/src/stream_list.ts

```diff
@@ -978,13 +978,14 @@ export function set_event_handlers({
         }
     });
 
-    $("#clear_search_stream_button").on("click", clear_search);
-
     $("#streams_header")
         .expectOne()
         .on("click", (e) => {
             e.preventDefault();
-            if (e.target.id === "streams_inline_icon") {
+            if (
+                e.target.id === "streams_inline_icon" ||
+                $(e.target).parent().hasClass("input-button")
+            ) {
                 return;
```

Comment:
Can you explain this change? I'm not sure what problem is being fixed here, but I wonder if just adding an actual class for what this click handler is supposed to be targeting, or using `stopPropagation` somewhere, would be better.

Reply:
In the left sidebar, the filter input is being added to the `#streams_header` element which stretches to accomodate it, which you can visualize using the images below:
| Before | After |
|--------|--------|
| <img width="322" alt="Screenshot 2025-07-03 at 5 42 12 AM" src="https://github.com/user-attachments/assets/4eae0636-6ce9-435b-bdc5-eaf9a91af487" /> | <img width="402" alt="Screenshot 2025-07-03 at 5 37 03 AM" src="https://github.com/user-attachments/assets/8b636a10-55f8-40d0-98f6-71eb1a10c387" /> |

Reply:
Clicking anywhere on the "#streams_header" element toggles the search bar according to the current logic. The previous logic checked if the user was clicking on the "Add channels" button, and preventing that event from toggling the filter input. I have included the clear input button to that same logic.